syntax = "proto2";

message User {
  required string username = 1;
  repeated string groupnames = 2;
}

message UserList {
    repeated User users = 1;
}

message Group {
  required string groupname = 1;
  repeated string usernames = 2;
}

message GroupList {
    repeated Group groups = 1;
}

message Message {
  enum Type {
    DIRECT = 1;
    GROUP = 2;
  }

  required Type type = 1;
  required string from = 2;
  optional string toUser = 3;
  optional string toGroup = 4;
  required string msg = 5;
}

message MessageList {
    repeated Message messages = 1;
}

message UserError {
  required string message = 1;
}
